File: objectiveai-rs/src/functions/inventions/recursive/response/streaming/object.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub enum Object {
    #[serde(rename = "alpha.scalar.function.invention.recursive.chunk")]
    AlphaScalarFunctionInventionRecursiveChunk,
    #[serde(rename = "alpha.vector.function.invention.recursive.chunk")]
    AlphaVectorFunctionInventionRecursiveChunk,
}

impl From<Object> for crate::functions::inventions::response::streaming::Object {
    fn from(object: Object) -> Self {
        match object {
            Object::AlphaScalarFunctionInventionRecursiveChunk => {
                Self::AlphaScalarFunctionInventionChunk
            }
            Object::AlphaVectorFunctionInventionRecursiveChunk => {
                Self::AlphaVectorFunctionInventionChunk
            }
        }
    }
}

File: objectiveai-js/src/functions/inventions/recursive/response/streaming/object.ts
import { z } from "zod";

export const FunctionsInventionsRecursiveResponseStreamingObjectSchema = z.enum(["alpha.scalar.function.invention.recursive.chunk","alpha.vector.function.invention.recursive.chunk"]).meta({ title: "functions.inventions.recursive.response.streaming.Object" });
export type FunctionsInventionsRecursiveResponseStreamingObject = z.infer<typeof FunctionsInventionsRecursiveResponseStreamingObjectSchema>;
